extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/VarBinaryChunkInputStreamGenerator.java

```diff
@@ -7,35 +7,184 @@
+
+                // do the write
+                activeChunk.copyFromTypedArray(b, off, activeChunkByteCount, writeLen);
+
+                // increment the counts
+                writtenTotalByteCount += writeLen;
+                activeChunkByteCount += writeLen;
+                off += writeLen;
+
+                remaining -= writeLen;
+
+                // allocate a new chunk when needed
+                if (activeChunkByteCount == BYTE_CHUNK_SIZE) {
+                    byteChunks.add(activeChunk = WritableByteChunk.makeWritableChunk(BYTE_CHUNK_SIZE));
+                    activeChunkByteCount = 0;
+                }
+            }
+        }
+
+        public long size() {
+            return writtenTotalByteCount;
+        }
+
+        /***
+         * computes the size of the payload from sPos to ePos (inclusive)
+         *
+         * @param sPos the first data item to include in this payload
+         * @param ePos the last data item to include in this payload
+         * @return number of bytes in the payload
+         */
+        public long getPayloadSize(int sPos, int ePos) {
+            return offsets.get(ePos + 1) - offsets.get(sPos);
+        }
+
+        /***
+         * write payload from sPos to ePos (inclusive) to the output stream
+         *
+         * @param dos the data output stream to populate with data
+         * @param sPos the first data item to include in this payload
+         * @param ePos the last data item to include in this payload
+         * @return number of bytes written to the outputstream
+         * @throws IOException if there is a problem writing to the output stream
+         */
+        public long writePayload(LittleEndianDataOutputStream dos, int sPos, int ePos) throws IOException {
+            final long writeLen = getPayloadSize(sPos, ePos);
+            long remainingBytes = writeLen;
+
+            long startBytePos = offsets.get(sPos);
+            while (remainingBytes > 0) {
+                final int chunkIdx = (int)(startBytePos / BYTE_CHUNK_SIZE);
+                final int byteIdx = (int)(startBytePos % BYTE_CHUNK_SIZE);
+
+                final ByteChunk<?> chunk = byteChunks.get(chunkIdx);
+
+                final int len = (int) Math.min(remainingBytes, BYTE_CHUNK_SIZE - byteIdx);
+
+                // do the write (using the stream adapter utility)
+                ByteChunkToOutputStreamAdapter.write(dos, chunk, byteIdx, len);
+
+                // increment the offsets
+                startBytePos += len;
+                remainingBytes -= len;
+            }
+            return writeLen;
+        }
+
+        @Override
+        public void close() {
+            try {
+                super.close();
+            } catch (IOException e) {
+                // ignore this error
+            }
+
+            // close the offset and byte chunks
+            if (offsets instanceof PoolableChunk) {
```

Comment:
`WriteableChunk`s are Poolable, you do not need the instanceof checks or the cast to PoolableChunk to call `close`.